var/spack/repos/builtin/packages/ccs-qcd/package.py

```diff
@@ -33,24 +19,33 @@ class CcsQcd(MakefilePackage):
     version('master', branch='master')
     version('1.2.1', commit='d7c6b6923f35a824e997ba8db5bd12dc20dda45c')
 
-    variant('class', values=int, default=1,
+    variant('class', default=1, values=('1', '2', '3', '4', '5', '6'),
```

Comment:
@alalazo @scheibelp @becker33 I believe we discovered a bug recently where values either can't be ints or can't be strings?

Reply:
I'm not sure of the bug you are referring to, might have missed that in the stream of notifications :thinking:  Here it seems to me we are also deleting the error message in the deleted function, is it intentional?

Reply:
class_validator was intentionally deleted, but I will modify it to use it.

Reply:
When values ​​is a list, the intended error message is not output.

Reply:
$ spack info ccs-qcd class=7
==> Error: unrecognized arguments: class=7

Reply:
I think that's because the default is an int but the values are strings.

Reply:
The intended error message is what is output by class_validator.
$ spack install ccs-qcd %gcc@8.3.1 class=7
==> Error: class: Choose one of the following:
1  - 8x8x8x32 (default MPI config: 1x1x1)
2  - 32x32x32x32 (default MPI config: 4x4x4)
3  - 64x64x64x32 (default MPI config: 8x8x8)
4  - 160x160x160x160 (default MPI config: 20x20x20)
5  - 256x256x256x256 (default MPI config: 32x32x32)
6  - 192x192x192x192 (default MPI config: 24x24x24)